packages/dht/src/connection/DeferredConnection.ts

```diff
@@ -0,0 +1,44 @@
+import { ConnectionType, IConnection } from './IConnection'
+import { EventEmitter } from 'events'
+import { ConnectionID } from '../types'
+import { PeerDescriptor } from '../proto/DhtRpc'
+import { Logger } from '../helpers/Logger'
+
+const logger = new Logger(module)
+
+export class DeferredConnection extends EventEmitter implements IConnection {
+    connectionId: ConnectionID
+    private buffer: Uint8Array[] = []
+    public connectionType = ConnectionType.DEFERRED
+    private readonly peerDescriptor: PeerDescriptor
+
+    constructor(private targetPeerDescriptor: PeerDescriptor) {
```

Comment:
Yes, but we already have `peerDescriptor` field, and we assign `this.peerDescriptor = targetPeerDescriptor`.  What now happens it that we have two separate fields in `DeferredConnection`:

- private peerDescriptor: PeerDescriptor (created explicitly at line 13)
- private targetPeerDescriptor: PeerDescriptor (created by TypeScript because of the `private` modifier)

Both have the same value. The value of `targetPeerDescriptor` field is never read.